pkg/controller/infrastructure/infraflow/reconcile.go

```diff
@@ -581,35 +666,45 @@ func (c *FlowContext) ensureZones(ctx context.Context) error {
 		tagsPrivate[TagKeyRolePrivateELB] = TagValueELB
 		desired = append(desired,
 			&awsclient.Subnet{
-				Tags:                        tagsWorkers,
-				VpcId:                       c.state.Get(IdentifierVPC),
-				CidrBlock:                   zone.Workers,
-				AvailabilityZone:            zone.Name,
-				AssignIpv6AddressOnCreation: ptr.To(false),
+				Tags:                                    tagsWorkers,
+				VpcId:                                   c.state.Get(IdentifierVPC),
+				AvailabilityZone:                        zone.Name,
+				AssignIpv6AddressOnCreation:             ptr.To(isIPv6(c.ipFamilies)),
+				Ipv6Native:                              ptr.To(!isIPv4(c.ipFamilies)),
+				EnableResourceNameDnsAAAARecordOnLaunch: ptr.To(!isIPv4(c.ipFamilies)),
 			},
 			&awsclient.Subnet{
 				Tags:                        tagsPublic,
 				VpcId:                       c.state.Get(IdentifierVPC),
-				CidrBlock:                   zone.Public,
 				AvailabilityZone:            zone.Name,
-				AssignIpv6AddressOnCreation: ptr.To(false),
+				AssignIpv6AddressOnCreation: ptr.To(isIPv6(c.ipFamilies)),
+				CidrBlock: func(cidr string) string {
+					if cidr == "" {
+						return "10.0.32.0/20"
```

Comment:
also here regarding the "magic IP"

Reply:
Same as above, it is probably better to let the user configure the CIDR for the public range.